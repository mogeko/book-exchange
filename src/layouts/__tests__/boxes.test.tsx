import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import Box from "@/layouts/boxes";

describe("Home", () => {
  it("renders a Display Box with Header", () => {
    const { container } = render(
      <Box title="Example Title">
        <></>
      </Box>
    );

    expect(screen.getByRole("heading")).toHaveTextContent("Example Title");

    expect(container).toMatchSnapshot();
  });

  it("renders a SubBox", () => {
    const { container } = render(
      <Box.SubBox title="Example SubBox">
        <></>
      </Box.SubBox>
    );
    expect(screen.getByRole("heading")).toHaveTextContent("Example SubBox");

    expect(container).toMatchSnapshot();
  });
});
