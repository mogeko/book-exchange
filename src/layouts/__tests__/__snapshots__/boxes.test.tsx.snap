// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders a Display Box with Header 1`] = `
<div>
  <div
    class="flex flex-col"
  >
    <h1
      class="text-2xl w-full py-2"
    >
      Example Title
    </h1>
  </div>
</div>
`;

exports[`Home renders a SubBox 1`] = `
<div>
  <div
    class="flex flex-col items-start my-2 prose"
  >
    <h2
      class="text-lg capitalize w-full my-2"
    >
      Example SubBox
    </h2>
  </div>
</div>
`;
